movabletype/movabletype
# Update extlibs

- CGI (4.36 to 4.38)
```
4.38 2017-12-01

    [ TESTING ]
    - command_line.t: Avoid -I for libs (GH #224, thanks to cpansprout)

4.37 2017-11-01

    [ FIX ]
    - Fix incorrect quoting of ? in ->url (GH #112, GH #222, with
      thanks to Reuben Thomas)
```

- CGI::Fast (2.12 to 2.13)
```
2.13 2017-11-17
    [TESTING]
    remove use of Test::Deep completely (GH #17)
```
- Class::Accessor (0.34 to 0.51)
```
0.50 2017-10-20
    Thanks to Jonas B. Nielsen for working through the RT queue.
    - patch for speed increase RT#84838
    - patch for faster constructor RT#57353
    - fixed typos for RT#61304 and RT#86422
```
- File::Copy::Recursive (0.38 to 0.40)
```
0.40 Tue Jan 16 10:00:09 2017
    - github Issue #5 - Deep directories pathmk
    - rt 123966 - switch to bsd_glob() since glob() will disappear in perl 5.30
    - rt 123971 - skip symlink tests when the OS does not support symlinks
    - rt 123970 - use Path::Tiny instead of File::Slurp in tests
    - rt 117241 - add test for read only directories

0.39 Fri Dec 29 13:26:12 2017
    - tidy code
    - Change into directory before emptying it
    - Stop emptying/removing a path if it is changed out from underneath us
    - pathrm() fixes
    - Actual unit tests!
```
- HTTP::Message (6.13 to 6.14)
```
6.14      2017-12-20 22:28:48Z
    - Add some useful examples in HTTP::Request (GH #92) (Chase Whitener).
      Batch requests are now explained.
    - PUT and PATCH docs updated (GH #84) (saturdaywalkers)
    - Trim trailing \r from status line so message() doesn't return it (GH #87) (Felipe Gasper)
    - Bring test coverage of HTTP::Config to 100% (GH #85) (Pete Houston)
    - Add 103 Early Hints to HTTP::Status (GH #94) (Tatsuhiko Miyagawa)
```
- JSON (2.96 to 2.97001)
```
2.97001 2017-12-21
    - updated backportPP with JSON::PP 2.97001

2.97000 2017-11-21
    - updated backportPP with JSON::PP 2.97000
    - use 5 digit minor version number for a while to avoid
      confusion
    - fixed is_bool to use blessed() instead of ref()
```
- JSON::PP (2.96 to 2.97001)
```
2.97001 2017-12-21
    - tweak internal number detector to always considers a flagged
      value as a string (GH#35, haarg++)
    - clarify json_pp options (RT-123766; Dan Jacobson)

2.97000 2017-11-21
    - fix is_bool to use blessed() instead of ref() (jwrightecs++)
    - use 5 digit minor version number for a while to avoid confusion
      (GH#33, dansut)
```
- Net::FTPSSL (0.38 to 0.39)
```
0.39 2017/10/23 08:30:00
  - Updated the LISCENSE file to 2017 (from 2015)
  - Changed default from TLS v1.0 to TLS v1.2 and rearranged the POD a bit.
  - Started depreciating useSSL.  Now prints out warning if used!  Warning
    says to use option "SSL_version" instead.
```
- Try::Tiny (0.28 to 0.30)
```
0.30      2017-12-21 07:23:03Z
  - expand "when" test skippage to more perl versions

0.29      2017-12-19 03:51:26Z
  - skip tests of "when" and "given/when" usage for perl 5.27.7 *only* (see
    RT#123908)
```
- URI (1.72 to 1.73)
```
1.73      2018-01-09 06:42:51Z
    - Update documentation for URI::_punycode (GH Issue #45)
```
- boolean (0.45 to 0.46)
```
0.46 Fri Jul  8 17:14:32 UTC 2016
 - Apply PR/14 @dagolden++
 - Adds CLONE method for thread safety
```
- LWP (6.26 to 6.31)
```
6.31      2017-12-11 01:55:53Z
    - fix version numbering (RT#123841)

6.30      2017-12-07
    - Use tr/// instead of s/// where appropriate (Ville Skyttﾃ､) (GH #265)
    - Use parent -norequire instead of base to not look for external
      dependencies (Fabian Zeindler) (GH #259)
    - Fix run_handlers to allow assigning to the request / response (Gianni
      Ceccarelli) (GH #274)

6.29      2017-11-06
    - Fix some version numbers

6.28      2017-11-06
    - Remove last use of Geopt::Std (Sergey Remanov) (GH #267)
    - Include unmatched connect error in status string (Patrik Lundin) (GH #269)
    - Fix insecure open FILEHANDLE,EXPR (Takumi Akiyama) (GH #270)

6.27      2017-09-21
    - Switch to Getopt::Long in lwp-download (GH #262)
    - Fix lwp-request -C (GH #261)
    - Hide LWP::Protocol::http::Socket, LWP::Protocol::http::SocketMethods and
      LWP::Debug::TraceHTTP::Socket from PAUSE
    - Add tests for the "get" & "head" functions (GH #252)
    - Update lwpcook.pod (GH #256)
    - Handle undefined values in ->credentials (GH #157)
    - Fix lwp-mirror options checks.
    - Update bin/ scripts to use $LWP::VERSION instead of ->Version()
    - Improve lwp-download --help (GH #262)
```
